src/targets/gpu/include/migraphx/gpu/code_object_op.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef MIGRAPHX_GUARD_AMDMIGRAPHX_CODE_OBJECT_OP_HPP
+#define MIGRAPHX_GUARD_AMDMIGRAPHX_CODE_OBJECT_OP_HPP
+
+#include <migraphx/config.hpp>
+#include <migraphx/value.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/functional.hpp>
+#include <migraphx/gpu/kernel.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+struct context;
+
+struct code_object_op
+{
+    value::binary code_object;
+    std::string symbol_name;
+    std::size_t global;
+    std::size_t local;
+    std::vector<shape> expected_inputs;
+    shape output;
+    kernel k;
```

Comment:
At some point we will want to be able to share the kernel for several code_object_op's since often in a model there are layers with identical dimensions and operations.

Reply:
> since often in a model there are layers with identical dimensions and operations.

After applying horizontal fusions, I don't really see that much in models. We can try to see how to share the same binaries across different operators although it may be a little complicated because of serialization.